### PR TITLE
W04 improvements - more linked list tests

### PR DESCRIPTION
**Pull request by ephraimkunz**

### CODE REVIEW

File: week04/code/LinkedListTester.cs [hunk 1]
```diff
@@ -16,12 +16,24 @@ public static class LinkedListTester {
         ll.InsertTail(-1);
         Console.WriteLine(ll.ToString()); // <LinkedList>{5, 4, 3, 2, 2, 2, 1, 0, -1};
 
+        var ll2 = new LinkedList();
+        ll2.InsertTail(1);
+        Console.WriteLine(ll2.HeadAndTailAreNotNull()); // True
+
```
Comment (ephraimkunz) [inline week04/code/LinkedListTester.cs:22]: Catches `InsertTail` not updating the head or tail pointer correctly.

File: week04/code/LinkedListTester.cs [hunk 2]
```diff
@@ -16,12 +16,24 @@ public static class LinkedListTester {
...
         Console.WriteLine(ll.ToString()); // <LinkedList>{5, 4, 3, 2, 2, 2, 1}
 
+        var ll3 = new LinkedList();
+        ll3.RemoveTail();
+        Console.WriteLine(ll3.ToString()); // <LinkedList>{}
+        ll3.InsertHead(2);
+        ll3.RemoveTail();
+        Console.WriteLine(ll3.ToString()); // <LinkedList>{}
+        Console.WriteLine(ll3.HeadAndTailAreNull()); // True
+
```
Comment (ephraimkunz) [inline week04/code/LinkedListTester.cs:36]: Catches `RemoveTail` not working properly on an empty list or a list with one element.

File: week04/code/LinkedListTester.cs [hunk 3]
```diff
@@ -41,6 +53,14 @@ public static class LinkedListTester {
...
         Console.WriteLine(ll.ToString()); // <LinkedList>{4, 35, 2, 2}
 
+        var ll4 = new LinkedList();
+        ll4.Remove(0);
+        Console.WriteLine(ll4.ToString()); // <LinkedList>{}
+        ll4.InsertHead(2);
+        ll4.Remove(2);
+        Console.WriteLine(ll4.ToString()); // <LinkedList>{}
+        Console.WriteLine(ll4.HeadAndTailAreNull()); // True
+
```
Comment (ephraimkunz) [inline week04/code/LinkedListTester.cs:63]: Catches `Remove` not working properly on an empty list or a list with 1 element.

**Review by sburton42 — APPROVED** [review]
Thank you!

**Pull request merged**